File: src/sort/js/gnomesort.js
function gnomeSort(list) {
    let pos = 0
    while (pos < list.length) {
        if (pos == 0 || list[pos] >= list[pos-1]) pos += 1 
        else {
            [ list[pos], list[pos-1] ] = [ list[pos-1], list[pos] ]
            pos -= 1
        }
    }
    return list
}

console.log(
    gnomeSort(process.argv.slice(2).map(function (x) { 
        return parseInt(x, 10); 
    })).join(" ")
)

File: src/sort/python/gnomesort.py
from sys import argv

def gnomeSort(list):
    pos = 0
    while pos < len(list):
        if (pos == 0) or (list[pos] >= list[pos-1]):
            pos += 1
        else:
            list[pos], list[pos-1] = list[pos-1], list[pos]
            pos -= 1
    return list

print(*gnomeSort([ int(x) for x in argv[1:] ]))